### DISCUSSION TITLE
Custom user list does not return data

### DISCUSSION THREAD

**Discussion by michalss** [henrikfroehling/Trakt.NET#427] [Help] [answered]
### Failure description?

```csharp
var user = await _client.Users.GetSettingsAsync();
var userName = "";
if (user != null)
{
  userName = user.Value.User.Username;
}
var usersFolder = await _client.Users.GetPersonalListsAsync(userName);
```

### Version

1.3.0 (Latest)

### Relevant Stacktrace

There is not any, just return empty list

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by michalss** [comment]
If i try pure api endpoint it does work ok..

``
{{API_URL}}/users/{user}/lists
``

**Comment by michalss** [comment]
Ok this is strange but List need to be public, but this is wrong it should also work if the list is private.

**Comment by henrikfroehling** [comment]
This looks like an authorization problem.
OAuth for [`{{API_URL}}/users/{user}/lists`](https://trakt.docs.apiary.io/#reference/users/lists/get-a-user's-personal-lists) is optional.
Did you set the `_client`'s authorization like described in the [documentation](https://henrikfroehling.github.io/Trakt.NET/docs/auth.html)?
Without authorization only public lists will be returned and with authorization public and private lists.

**Comment by michalss** [comment]
im always authorized with token an it is valid... check the code, If i would not be auth i would never be able to read my own Username
This would not
```csharp
var user = await _client.Users.GetSettingsAsync();
```
work otherwise

**Comment by henrikfroehling** [comment, marked as answer]
Sorry, you're right. I missed the line.

Did you try setting `_client.Configuration.ForceAuthorization = true`?

**Comment by michalss** [comment]
works with force, but i dont get why this is needed..

**Comment by henrikfroehling** [comment]
The Trakt API has endpoints that require OAuth authorization and endpoints where authorization is optional.
The endpoints where authorization is optional are not used in the library with forced authorization.
Instead, authorization is treated as opt-in.
But basically, it doesn't matter which behavior is used as the default.
It's just a decision made that the default is what is specified in the Trakt API documentation.
If you want to explicitly use authorization, enable the configuration flag.
If it were the other way around, you would still need to set a configuration flag for optional authorization.

Docs => https://henrikfroehling.github.io/Trakt.NET/docs/configuration.html#forced-authorization
